verification-annotations/src/crux.rs

```diff
@@ -10,17 +10,31 @@
 // FFI wrapper for Crux-mir static simulator tool
 /////////////////////////////////////////////////////////////////
 
-// Create an abstract value of type <T>
-//
-// This should only be used on types that occupy contiguous memory
-// and where all possible bit-patterns are legal.
-// e.g., u8/i8, ... u128/i128, f32/f64
-pub fn abstract_value<T: crucible::Symbolic>() -> T {
-    // We assume the string argument is just for reporting to the user, and
-    // doesn't affect the results.
-    T::symbolic("")
+use crate::traits::*;
+
+impl <T: AbstractValue> VerifierNonDet for T {
+    fn verifier_nondet(self) -> Self {
+        T::abstract_value()
+    }
+}
+
+impl <T: crucible::Symbolic> AbstractValue for T {
+    fn abstract_value() -> Self {
+        // We assume the string argument is just for reporting to the user, and
+        // doesn't affect the results.
+        let r : T = crucible::Symbolic::symbolic("");
+        r
+    }
 }
 
+impl <T: crucible::Symbolic + Default> Symbolic for T {
+    fn symbolic(_desc: &'static str) -> Self {
```

Comment:
Maybe pass str to symbolic below, instead of ""?

Reply:
oops! Good catch.